### PR TITLE
OAK-9169 - Remove remote elastic indexes when index definition is rem…

### PR DESCRIPTION
**Pull request by averma21**
…oved

* Checking existing local elastic indexes before fetching remote index

### CODE REVIEW

**Review by fabriziofortino — APPROVED** [review]

**Review by fabriziofortino — COMMENTED** [review]
I had a second thought on this. If there is one elastic index definition and then it gets removed (`existingIndices` will be empty) we won't remove the index in elastic.

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexProviderService.java [hunk 1]
```diff
@@ -177,7 +180,11 @@ private void deactivate() {
         }
     }
 
-    private void registerIndexCleaner(Config contextConfig) {
+    private void registerIndexCleaner(Config contextConfig) throws IOException {
+        boolean reachable = elasticConnection.getClient().ping(RequestOptions.DEFAULT);
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexProviderService.java]: Can we move this logic inside a method in `ElasticConnection`? Something like 

```
public boolean isAvailable() { ... }
```

Comment (averma21) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexProviderService.java]: added.

**Review by fabriziofortino — APPROVED** [review]

**Review by chibulcu — APPROVED** [review]

**Comment by averma21** [comment]
Committed to trunk. Closing.

**Pull request closed without merging**